vertx-sql-client/src/main/java/io/vertx/sqlclient/SqlConnection.java

```diff
@@ -88,5 +89,7 @@
    * @param handler the completion handler
    */
   void close(Handler<AsyncResult<Void>> handler);
+  
+  DatabaseMetaData getDatabaseMetaData();
```

Comment:
rename to `databaseMetadata()` - I'm wondering if that should not be directly on `SqlClient`

Reply:
Usually exposing such metadata is an internal feature for those users, I think it's better to keep the `SqlClient` API simple.

Reply:
The SqlClient is more focused on one-shot operations, whereas obtaining DB metadata isn't really a one-shot operation, it's mainly something that would make sense to obtain after you've obtained a connection (like in JDBC).
Also, I'm considering the new metadata class as an SPI, which also supports @BillyYccc 's comment about keeping it off of the "golden path" API usage